test/NuGet.Core.Tests/NuGet.Common.Test/MigrationRunnerTests.cs

```diff
@@ -62,5 +62,35 @@ public void Run_WhenExecutedInParallelThenOnlyOneMigrationFileIsCreated_Success(
             Assert.Equal(1, files.Length);
             Assert.Equal(Path.Combine(directory, "1"), files[0]);
         }
+
+        [Fact]
+        public void Run_WhenAThreadAbandonsMutexThenNextMigrationRunReleasesMutexAndCreatesMigrationFile_Success()
+        {
+            Mutex _orphan = new Mutex(false, "NuGet-Migrations");
+
+            // Arrange
+            Thread t = new Thread(new ThreadStart(AbandonMutex));
+            t.Start();
+            t.Join();
+
+            string directory = MigrationRunner.GetMigrationsDirectory();
+            if (Directory.Exists(directory))
+                Directory.Delete(path: directory, recursive: true);
+
+            // Act
```

Comment:
Should call `t.Abort();` to simulate actual abortion/termination of thread which holding the mutex?

Reply:
No need because the local function `AbandonMutex` exits without invoking `ReleaseMutex()` on the mutex object.

Reply:
@erdembayar just FYI, `Thread.Abort` has always been problematic in .NET, and since .NET Core 1.0 it's obsolete, and in fact throws if you try to call it: https://learn.microsoft.com/en-us/dotnet/api/system.threading.thread.abort?view=net-7.0#system-threading-thread-abort